bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -398,6 +365,60 @@ pub mod pallet {
 				.reanchored(&T::BridgeHubLocation::get(), &T::UniversalLocation::get())
 				.map_err(|_| Error::<T>::LocationConversionFailed)
 		}
+
+		fn swap_fee_asset_and_burn(
+			who: AccountIdOf<T>,
```

Comment:
The flow throughout this file keeps turning Location to AccountId back and forth - I suggest changing to Location here and using the original Location everywhere a Location is needed and do `.into()` where an account is needed. I think it will simplify and clarify some of the code.

```suggestion
			who: Location,
```